tools/BaseImageUpdate/baseImage.ps1

```diff
@@ -0,0 +1,86 @@
+# Run this on E2E test agent in PWSH
+function Setup-BaseImage-Script
+{
+    # Change current directory to ../iotedge
+    try
+    {
+        $(cd $(Build.SourcesDirectory)) 
+    }
+    catch
+    {
+        $curDir = pwd
+        $iotedgeDir=$($curDir -split("iotedge", 2))
+        $iotedgeDir = $iotedgeDir[0] + "iotedge"
+        cd $iotedgeDir
+    }
```

Comment:
Yup. It seems like `$env:BUILD_SOURCESDIRECTORY` is not defined neither $PSScriptRoot

Reply:
`$env:BUILD_SOURCESDIRECTORY` is only defined in Azure pipelines builds. `$PSScriptRoot` is an automatic variable that contains the directory from which a script is being run, since PowerShell 3.0. We're already using this logic in other scripts.

Reply:
> $PSScriptRoot is an automatic variable that contains the directory from which a script is being run

Ah I see. That's why it's returning empty when I tried running it striaght on the pwsh.

Reply:
Oh right, it's only available inside a script :)